front/src/common/BootstrapSNCF/InputSNCF.tsx

```diff
@@ -115,44 +104,48 @@ const InputSNCF = ({
   };
 
   // Displays a button at the end of the input field to clear the input
-  const addClearButton = () => {
-    const displayClearButton = clearButton && value;
-
-    // Returns null if the clear button is not used
-    if (!displayClearButton) return null;
-
+  const addClearButton = () => (
     // Else renders the button
-    return (
-      <button type="button" className="btn-clear btn-primary" onClick={onClear}>
-        <span className="sr-only">Supprimer le texte</span>
-        <i className="icons-close" aria-hidden="true" />
-      </button>
-    );
-  };
-
+    <button type="button" className="btn-clear btn-primary" onClick={onClear}>
+      <span className="sr-only">Supprimer le texte</span>
+      <i className="icons-close" aria-hidden="true" />
+    </button>
+  );
   // Renders a basic input field without any underlying list
   const basicInput = () => {
     const inputComponent = (
       <>
         {label && (
           <label
-            className={flex ? 'font-weight-medium mb-0 mr-2' : 'font-weight-medium mb-2'}
+            className={cx('font-weight-medium', { 'mb-0 mr-2': isFlex, 'mb-2': !isFlex })}
             htmlFor={id}
           >
             {label}
           </label>
         )}
-        <div className={appendOptions ? 'input-group' : ''}>
-          <div className={`form-control-container ${invalidClass} ${unit ? 'has-right-icon' : ''}`}>
+        <div className={cx({ 'input-group': appendOptions })}>
+          <div
+            className={cx('form-control-container', {
+              'is-invalid': isInvalid,
+              'has-right-icon': unit,
+            })}
+          >
             <input
+              autoFocus={focus}
               type={type}
               onChange={onChange}
-              className={`form-control ${backgroundColor} ${formSize} ${readOnlyFlag} ${clearOption} ${condensedInput} ${textAlignmentClass}`}
+              className={cx('form-control', textAlignmentClass, {
+                'bg-white': whiteBG,
+                'clear-option': clearButton,
```

Comment:
Otherwise, the placeholder (if it's long) is partially hidden by this class
```suggestion
                'clear-option': clearButton && value,
```